libs/ledger-live-common/src/families/cosmos/crypto/Cosmos.ts

```diff
@@ -0,0 +1,159 @@
+import { makeLRUCache } from "../../../cache";
+import network from "../../../network";
+import { CosmosRewardsState } from "../types";
+import cosmosBase from "./cosmosBase";
+import { parseUatomStrAsAtomNumber } from "../logic";
+
+class Cosmos extends cosmosBase {
+  lcd: string;
+  stakingDocUrl: string;
+  unbonding_period: number;
+  constructor() {
+    super();
+    this.unbonding_period = 21;
+    this.lcd = "https://cosmoshub4.coin.ledger.com";
+    this.stakingDocUrl =
+      "https://support.ledger.com/hc/en-us/articles/360014339340-Earn-Cosmos-ATOM-staking-rewards-in-Ledger-Live?docs=true";
+    this.min_gasprice = 0.025;
+  }
+  private computeAvgYearlyInflation = (rewardsState: CosmosRewardsState) => {
+    // Return invalid rewardsState if
+    // rewardsState.currentValueInflation is not between inflationMinRate and inflationMaxRate
+    const inflationSlope =
+      (1 - rewardsState.actualBondedRatio / rewardsState.targetBondedRatio) *
+      rewardsState.inflationRateChange;
+    const unrestrictedEndOfYearInflation =
+      rewardsState.currentValueInflation * (1 + inflationSlope);
+
+    if (
+      unrestrictedEndOfYearInflation <= rewardsState.inflationMaxRate &&
+      unrestrictedEndOfYearInflation >= rewardsState.inflationMinRate
+    ) {
+      return (
+        (rewardsState.currentValueInflation + unrestrictedEndOfYearInflation) /
+        2
+      );
+    }
+
+    if (unrestrictedEndOfYearInflation > rewardsState.inflationMaxRate) {
+      const diffToMax =
+        rewardsState.inflationMaxRate - rewardsState.currentValueInflation;
+      const maxPoint = diffToMax / inflationSlope;
+      const averageInflation =
+        (1 - maxPoint / 2) * rewardsState.inflationMaxRate +
+        (maxPoint / 2) * rewardsState.currentValueInflation;
+      return averageInflation;
+    }
+
+    if (unrestrictedEndOfYearInflation < rewardsState.inflationMinRate) {
+      const diffToMin =
+        rewardsState.currentValueInflation - rewardsState.inflationMinRate;
+      const minPoint = diffToMin / inflationSlope;
+      const averageInflation =
+        (1 - minPoint / 2) * rewardsState.inflationMinRate +
+        (minPoint / 2) * rewardsState.currentValueInflation;
+      return averageInflation;
+    }
+
+    throw new Error("Unreachable code");
+  };
+  public getRewardsState(): any {
```

Comment:
can you type what is returned here ?
With a type like

```ts
type RewardState = {
 targetBondedRatio: number;
 communityPoolCommission: number;
 ....
};
```

Can be implemented in base.ts and then imported in all other files

Reply:
fixed